registry/token.go

```diff
@@ -51,10 +51,12 @@ func getToken(username, password string, params map[string]string, registryEndpo
 		reqParams.Add("scope", scopeField)
 	}
 
-	reqParams.Add("account", username)
+	if username != "" {
+		reqParams.Add("account", username)
+		req.SetBasicAuth(username, password)
+	}
 
 	req.URL.RawQuery = reqParams.Encode()
-	req.SetBasicAuth(username, password)
```

Comment:
I think I've figured out the problem. My fix exposed another bug when pulling all tags from a v2 registry. It gets back an empty list of tags (because the image doesn't exist), then pulls each of these tags - except it actually does nothing because it did not get back any tags at all!